### DISCUSSION TITLE
Automatic port forwarding failing in Codespaces

### DISCUSSION THREAD

**Discussion by valhuber** [community/community#30720] [Codespaces]
Per [this doc](https://docs.github.com/en/codespaces/developing-in-codespaces/forwarding-ports-in-your-codespace#about-forwarded-ports), I thought that printing a URL to the console would automatically create a port.  (I know I can create it manually - that works fine).

My app starts and prints this (note it includes __Running on http://localhost:5656__) :
```
Server starting on Codespaces -- explore sample data and API at:
..  https://valhuber-apilogicserver-apilogicproject-5694997xh4prw-5656.githubpreview.dev/

 * Serving Flask app "API Logic Server" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Running on http://localhost:5656/ (Press CTRL+C to quit)
API Logic Server - redirect /admin-app/index.html
127.0.0.1 - - [25/Aug/2022 00:36:15] "GET / HTTP/1.1" 302 -
```

Am I missing something?

**Comment by aishwaryabh** [comment]
Hi @valhuber! So printing the url allows you to follow the link to the console but you would still have to run the actual process for forwarding (like `npm start` for example). Also that port should automatically forward and show up on the ports tab once you run the process itself. What's your process for automatically forwarding the port?

**Comment by valhuber** [comment]
Hi, @aishwaryabh!

Thanks for getting back.  I appreciate it, because I really want to make this work.  The idea I discussed with Craig was to use Codespaces as the eval path for my [open source project](https://valhuber.github.io/ApiLogicServer/).

The [setup instructions](https://github.com/ApiLogicServer/ApiLogicProject#1-open-in-codespaces) aren't terribly burdensome, but this is about making them better.  In particular, I am experimenting in omitting the "define a port" discussion (to simplify things for devs).  

If you follow the setup instructions to launch codespaces, you will land in a Python / Flask project.

You run the first Launch Config (ApiLogicServer), which launches the server on localhost.  The main code is `api_logic_server_run.py`.  You will see that around line 434 is where I start the server; line 428 confirms that it's being launched on `localhost`.  I presumed that would trigger the automatic port forwarding...

> So, if I understood your "start the process" note, I think I did...

As I said, I am eager to make this work, so would be happy to screen share if these instructions are not clear.  I on US West Coast, and usually able to work weekends if that makes it easier for you.

**Comment by Th3S4mur41** [comment]
I had a similar issue with vite, where automatic port forwarding stopped working at some point...

The issue was that vite by default was only listening to localhost.
Even manually forwarding the port didn't work because it would forward it to 127.0.0.1 which vite wasn't listening on.

[Setting `server.host` to `true` in the vite config](https://github.com/Th3S4mur41/th3s4mur41.me/commit/511935cc5a0573b3bb44d4b573e5874804dd36b1#diff-58e6f63d87181b1c6a8cb6e5f1691df04aa32854456efcd52ca71c8541375d26) led to the server also listening on the local IP, and not only localhost which solved the issue.
